mlflow/utils/databricks_utils.py

```diff
@@ -418,8 +423,23 @@ def _fail_malformed_databricks_auth(profile):
         "https://github.com/databricks/databricks-cli." % profile
     )
 
+# Helper function to attempt to read OAuth Token from
+# mounted file in Databricks Model Serving environment
+def _get_model_dependency_oauth_token(should_retry=True):
+    try:
+        with open(MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH, "r") as f:
+            json_data = f.read()
+            oauth_dict = json.loads(json_data)
+            return oauth_dict["OAUTH_TOKEN"][0]["oauthTokenValue"]
+    except Exception as e:
+        if should_retry:
+            return _get_model_dependency_oauth_token(should_retry=False)
+        else:
+            raise e
+
+
 
-def get_databricks_host_creds(server_uri=None):
+def get_databricks_host_creds(server_uri=None, get_creds_for_model_serving_dep=False):
```

Comment:
can we make a modification to API parameters here? we will need this new flag to ensure we do not try to fetch limited scope OAuth token if user has some DIY workflow that calls `get_databricks_host_creds` for creds in model serving for something beyond just model dependencies.

Reply:
NOTE: in model serving environment, by default credential env vars and databricks cfg will not be populated but some customers may choose to manually configure environment variable to be read. Options here are 
a) support a new method for fetching oauth tokens in model serving environment 
b) bake logic into existing function if serving environment is on and no additional arg (calls to `get_databricks_host_creds` for other use cases in model serving won't work)
c) bake logic into existing function with arg and default to standard behavior if failures with oauth

Reply:
@victorsun123 What do you propose the behavior should be when someone uses the MLflow Deployments client to query an external model within a serving endpoint? MLflow Deployments uses `get_databricks_host_creds()`, and it *should* use the OAuth token in that case. How do we handle this?

Reply:
cc @aakrati ^

Reply:
Couple of points: 
* If we use, `get_creds_for_model_serving_dep` as a parameter how do we pass it in for RAG Models in mlflow deployments and vector search sdk? 

One way to do this would be to 
1. use environment variable to determine its a Databricks Model serving environment and populate a different DATABRICKS_OAUTH_TOKEN, this takes lower precedence than DATABRICKS_TOKEN
2. Update Mlflow deployments so then by default if user wants to use get_databricks_host_creds their code will set DATABRICKS_TOKEN but if nothing is set we use DATABRICKS_OAUTH_TOKEN (or perhaps call it INTERNAL_DATABRICKS_OAUTH_TOKEN)


`@dbczumar - What do you propose the behavior should be when someone uses the MLflow Deployments client to query an external model within a serving endpoint? MLflow Deployments uses get_databricks_host_creds(), and it should use the OAuth token in that case. How do we handle this?`

For external models and custom model endopints, we are thinking to use OAuth token as well, to keep tokens we use to do auth unified. OAuth tokens can encode information about resources, hence they can be used to encode multiple resources, like a Open AI chat model + a custom model can all be encoded within same OAuth token. The only downside I see is token can grow in size quite a bit with each resource, so at some point of time with very large number of dependency we might need sub tokens for each dependency. 
WDYT?

Reply:
@aakrati I think what you've described for having a precedence order with `DATABRICKS_TOKEN` and `INTERNAL_DATABRICKS_OAUTH_TOKEN` sounds great and should work generally for all MLflow APIs in serving, not just MLflow Deployments client. I think we should go with that approach.

Reply:
Actually, did some thinking there shouldn't be any problem with just trying to read from oauth file by default since in model serving environment, environment variable can be configured for reading token but databricks cfg shouldn't be populated. As such, this shouldn't break any existing workflow if we just try to get credentials from oauth token file